### ISSUE TITLE
Potentially swapped arguments in multipole gravity BCs

### ISSUE THREAD

**Issue by yut23** [AMReX-Astro/Castro#2421]
clang-tidy discovered that we call `multipole_symmetric_add` in `Gravity::fill_multipole_BCs` with the qL* and qU* arguments swapped:
https://github.com/AMReX-Astro/Castro/blob/519fc54577195b1774e559343b4ce6984344a239/Source/gravity/Gravity.cpp#L1963-L1966 https://github.com/AMReX-Astro/Castro/blob/519fc54577195b1774e559343b4ce6984344a239/Source/gravity/Gravity_util.H#L201-L212

This looks to have been the case since that code was originally introduced in https://github.com/AMReX-Astro/Castro/commit/c0ebbdb3b9d159fdbb40dd964fb5f7d808fd6338.

**Comment by zingale** [comment]
this appears to apply only to symmetry BCs that are not axisymmetry.

I think the way to test is to try the cubic sphere test with reflecting BCs.

**Comment by yut23** [comment]
Okay, I tried out `uniform_cube_sphere` with multipole BCs enabled and the special casing for the reflecting boundary being at `center` commented out in `Gravity::init_multipole_grav`. Swapping the arguments makes the version with a reflecting BC in x match the version with the full domain (the error is 0.367 without the swap, 0.008 with the swap, and 0.008 for the full domain).

For reference, I ran `./Castro3d.gnu.ex inputs gravity.direct_sum_bcs=0` for the full domain and `./Castro3d.gnu.ex inputs gravity.direct_sum_bcs=0 geometry.prob_lo="0 -1.6 -1.6" amr.n_cell="8 16 16" castro.lo_bc="3 2 2"` for the reflected domain.

Interestingly, I get a similar result using direct sum BCs as I do using multipole BCs without the swap, so there might be something off there too.
